File: JavaScript/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let answer;
    obj = {};
    for (let s of completion){
        if (obj[s]) {
            obj[s] += 1;
        } else {
            obj[s] = 1;
        }
    }
    for (let mem of participant){
        if (obj[mem]){
            obj[mem] -= 1;
        } else {
            answer = mem;
        }
    }
    return answer;;
}

File: C++/프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = ""; 
    map<string, int> s;
    for (auto mem: completion){
        s[mem] += 1;
    }
    for (auto mem: participant){
        if (s[mem] == 0){
            answer = mem;
            break;
        }
        s[mem] -= 1;
    }
    return answer;
}